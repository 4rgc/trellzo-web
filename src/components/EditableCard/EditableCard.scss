.card.editable-card {
	justify-content: space-around;

	.title {
		margin-bottom: 4px;
	}

	.card-container-no-text {
		padding: 2px;
		margin: 8px;
	}

	.btn-container {
		display: flex;
		flex-direction: row;
		justify-content: space-evenly;
		margin-top: 5px;
	}
}
.editable-card .title {
	margin-bottom: 4px;
}
