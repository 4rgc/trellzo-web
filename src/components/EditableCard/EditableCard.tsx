import Card from '../Card';
import TextInput from '../TextInput';
import './EditableCard.scss';

interface IEditableCardProps {
	bodyPlaceholder?: string;
	bodyValue?: string;
	onTitleChange?: (text: string) => void;
	onBodyChange?: (text: string) => void;
}

type EditableCardPropsWithoutTitle = IEditableCardProps & {
	hasTitleField?: false;
};
type EditableCardPropsWithTitle = IEditableCardProps & {
	hasTitleField?: true;
	titlePlaceholder?: string;
	titleValue?: string;
};

export type EditableCardProps = Omit<
	React.ComponentProps<typeof Card>,
	| 'isImageDisabled'
	| 'imageUrl'
	| 'content'
	| 'title'
	| 'onClick'
	| 'isDisabled'
> &
	(EditableCardPropsWithTitle | EditableCardPropsWithoutTitle);

const EditableCard: React.FC<EditableCardProps> = (props) => {
	const {
		bodyPlaceholder,
		bodyValue,
		onTitleChange,
		onBodyChange,
		...otherProps
	} = props;

	return (
		<Card
			{...otherProps}
			isImageDisabled={true}
			isDisabled={false}
			className="editable-card"
		>
			{props.hasTitleField && (
				<TextInput
					variant="single"
					value={props.titleValue}
					placeholder={props.titlePlaceholder}
					onChange={props.onTitleChange}
				/>
			)}
			<TextInput
				variant="multi"
				value={bodyValue}
				placeholder={bodyPlaceholder}
				onChange={onBodyChange}
				rows={14}
			/>
		</Card>
	);
};

export default EditableCard;
