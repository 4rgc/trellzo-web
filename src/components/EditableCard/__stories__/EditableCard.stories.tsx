import { ComponentMeta, ComponentStory } from '@storybook/react';
import EditableCard from '..';

export default {
	title: 'Components/EditableCard',
} as ComponentMeta<typeof EditableCard>;
