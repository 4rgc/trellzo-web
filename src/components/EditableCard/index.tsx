import EditableCard from './EditableCard';

export default EditableCard;
