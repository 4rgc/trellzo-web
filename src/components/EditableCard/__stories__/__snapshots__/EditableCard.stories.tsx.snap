// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/EditableCard Default smoke-test 1`] = `
<div class="card card-md card-noimage editable-card"
     style="width: 200px;"
>
  <div class="card-container-no-text">
    <textarea class="tinput-md"
              placeholder
              rows="14"
    >
    </textarea>
  </div>
</div>
`;

exports[`Components/EditableCard Large smoke-test 1`] = `
<div class="card card-lg card-noimage editable-card"
     style="width: 200px;"
>
  <div class="card-container-no-text">
    <textarea class="tinput-md"
              placeholder
              rows="26"
    >
    </textarea>
  </div>
</div>
`;

exports[`Components/EditableCard LargeWithTitle smoke-test 1`] = `
<div class="card card-lg card-noimage editable-card"
     style="width: 200px;"
>
  <div class="card-container-no-text">
    <input class="tinput-md title"
           value
    >
    <textarea class="tinput-md"
              placeholder
              rows="24"
    >
    </textarea>
  </div>
</div>
`;

exports[`Components/EditableCard Medium smoke-test 1`] = `
<div class="card card-md card-noimage editable-card"
     style="width: 200px;"
>
  <div class="card-container-no-text">
    <textarea class="tinput-md"
              placeholder
              rows="14"
    >
    </textarea>
  </div>
</div>
`;

exports[`Components/EditableCard Small smoke-test 1`] = `
<div class="card card-sm card-noimage editable-card"
     style="width: 200px;"
>
  <div class="card-container-no-text">
    <textarea class="tinput-md"
              placeholder
              rows="3"
    >
    </textarea>
  </div>
</div>
`;

exports[`Components/EditableCard SmallWithAButton smoke-test 1`] = `
<div class="card card-sm card-noimage editable-card"
     style="width: 200px;"
>
  <div class="card-container-no-text">
    <textarea class="tinput-md"
              placeholder
              rows="1"
    >
    </textarea>
    <div class="btn-container">
      <button class="btn-prm-md-thn ">
        click me
      </button>
    </div>
  </div>
</div>
`;

exports[`Components/EditableCard SmallWithTitle smoke-test 1`] = `
<div class="card card-sm card-noimage editable-card"
     style="width: 200px;"
>
  <div class="card-container-no-text">
    <textarea class="tinput-md"
              placeholder
              rows="3"
    >
    </textarea>
  </div>
</div>
`;

exports[`Components/EditableCard WithAButton smoke-test 1`] = `
<div class="card card-md card-noimage editable-card"
     style="width: 200px;"
>
  <div class="card-container-no-text">
    <textarea class="tinput-md"
              placeholder
              rows="12"
    >
    </textarea>
    <div class="btn-container">
      <button class="btn-prm-md-thn ">
        click me
      </button>
    </div>
  </div>
</div>
`;

exports[`Components/EditableCard WithTitle smoke-test 1`] = `
<div class="card card-md card-noimage editable-card"
     style="width: 200px;"
>
  <div class="card-container-no-text">
    <input class="tinput-md title"
           value
    >
    <textarea class="tinput-md"
              placeholder
              rows="12"
    >
    </textarea>
  </div>
</div>
`;

exports[`Components/EditableCard WithTwoButtons smoke-test 1`] = `
<div class="card card-md card-noimage editable-card"
     style="width: 200px;"
>
  <div class="card-container-no-text">
    <textarea class="tinput-md"
              placeholder
              rows="12"
    >
    </textarea>
    <div class="btn-container">
      <button class="btn-sec-sm-thn ">
        cancel
      </button>
      <button class="btn-prm-sm-thn ">
        ok
      </button>
    </div>
  </div>
</div>
`;
